hermes/probe/monitor_test.go

```diff
@@ -0,0 +1,119 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Authors:
+// - Alicja Kwiecinska (kwiecinskaa@google.com) github: alicjakwie
+// - Evan Spendlove, GitHub: evanSpendlove.
+//
+// Monitor_test tests the functions used in the monitor probe, as well
+// as the probes integration with Cloudprober.
+
+// Package probe implements the probe that Hermes uses to monitor
+// a storage system.
+package probe
+
+import (
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/probes/options"
+
+	metricpb "github.com/google/cloudprober/metrics/proto"
+	probes_configpb "github.com/google/cloudprober/probes/proto"
+	monitorpb "github.com/googleinterns/step224-2020/config/proto"
+)
+
+// genTestProbeConf generates a test HermesProbeDef proto config for
+// initialising and running a MonitorProbe.
+// Arguments:
+//	- name: pass the name for this probe instance.
+// Returns:
+//	- probeDef: a test probe config with a HermesProbeDef extension set.
+func genTestConfig(name string) (*probes_configpb.ProbeDef, *monitorpb.HermesProbeDef) {
+	probeDef := &probes_configpb.ProbeDef{
+		Name: proto.String(name),
+		Type: probes_configpb.ProbeDef_EXTENSION.Enum(),
+	}
+
+	hermesExtension := &monitorpb.HermesProbeDef{
+		ProbeName:    proto.String(name),
+		Targets:      []*monitorpb.Target{},
+		TargetSystem: monitorpb.HermesProbeDef_GCS.Enum(),
+		IntervalSec:  proto.Int32(3600),
+		TimeoutSec:   proto.Int32(60),
+		ProbeLatencyDistribution: &metricpb.Dist{
+			Buckets: &metricpb.Dist_ExplicitBuckets{
+				ExplicitBuckets: "0.1, 0.2, 0.4, 0.6 0.8, 1.6, 3.2, 6.4, 12.8, 1000",
+			},
+		},
+		ApiCallLatencyDistribution: &metricpb.Dist{
+			Buckets: &metricpb.Dist_ExplicitBuckets{
+				ExplicitBuckets: "0.000000002, 0.000000004, 0.000000008, 0.000000016, 0.00000032, 0.000000064, 0.000000128, 100",
+			},
+		},
+	}
+	proto.SetExtension(probeDef, monitorpb.E_HermesProbeDef_HermesProbeDef, hermesExtension)
+	return probeDef, hermesExtension
+}
+
+// genOptsFromConfig generates the options for the MonitorProbe
+// from the probe config passed.
+// Arguments:
+//	- t: testing object for throwing errors
+//	- cfg: probe config.
+// Returns:
+//	- opts: options for a probe generated from the probe config.
+func genOptsFromConfig(t *testing.T, cfg *monitorpb.HermesProbeDef) *options.Options {
+	opts := &options.Options{
+		Interval: time.Duration(cfg.GetIntervalSec()) * time.Second,
+		Timeout:  time.Duration(cfg.GetTimeoutSec()) * time.Second,
+	}
+
+	var err error
+	if opts.Logger, err = logger.NewCloudproberLog(cfg.GetProbeName()); err != nil {
+		t.Fatalf("error in initializing logger for the probe (%s): %v", cfg.GetProbeName(), err)
+	}
+
+	opts.ProbeConf = cfg
+
+	return opts
+}
+
+func TestInit(t *testing.T) {
+	mp := &MonitorProbe{}
+	_, hermesConf := genTestConfig("testProbe")
+	mp.Init("testProbe", genOptsFromConfig(t, hermesConf))
```

Comment:
Add some checks.
At the minimum, check the error returned by Init().

Reply:
I will add full checks to this test.

Reply:
I will add full checks to this test.